--- orig/lib/extract.c
+++ mod/lib/extract.c
@@ -52,6 +52,7 @@
 	ut.modtime = ut.actime = th_get_mtime(t);
 
 	/* change owner/group */
+#ifndef WIN32
 	if (geteuid() == 0)
 #ifdef HAVE_LCHOWN
 		if (lchown(filename, uid, gid) == -1)
@@ -79,7 +80,7 @@
 #endif
 		return -1;
 	}
-
+#endif /* WIN32 */
 	/* change permissions */
 	if (!TH_ISSYM(t) && chmod(filename, mode) == -1)
 	{
@@ -104,7 +105,11 @@
 	{
 		struct stat s;
 
+#ifdef WIN32
+		if (stat(realname, &s) == 0 || errno != ENOENT)
+#else
 		if (lstat(realname, &s) == 0 || errno != ENOENT)
+#endif
 		{
 			errno = EEXIST;
 			return -1;
@@ -313,7 +318,9 @@
 #ifdef DEBUG
 	printf("  ==> extracting: %s (link to %s)\n", filename, linktgt);
 #endif
+#ifndef WIN32
 	if (link(linktgt, filename) == -1)
+#endif
 	{
 #ifdef DEBUG
 		perror("link()");
@@ -348,7 +355,9 @@
 	printf("  ==> extracting: %s (symlink to %s)\n",
 	       filename, th_get_linkname(t));
 #endif
+#ifndef WIN32
 	if (symlink(th_get_linkname(t), filename) == -1)
+#endif
 	{
 #ifdef DEBUG
 		perror("symlink()");
@@ -386,8 +395,10 @@
 	printf("  ==> extracting: %s (character device %ld,%ld)\n",
 	       filename, devmaj, devmin);
 #endif
+#ifndef WIN32
 	if (mknod(filename, mode | S_IFCHR,
 		  compat_makedev(devmaj, devmin)) == -1)
+#endif
 	{
 #ifdef DEBUG
 		perror("mknod()");
@@ -425,8 +436,10 @@
 	printf("  ==> extracting: %s (block device %ld,%ld)\n",
 	       filename, devmaj, devmin);
 #endif
+#ifndef WIN32
 	if (mknod(filename, mode | S_IFBLK,
 		  compat_makedev(devmaj, devmin)) == -1)
+#endif
 	{
 #ifdef DEBUG
 		perror("mknod()");
@@ -461,7 +474,11 @@
 	printf("  ==> extracting: %s (mode %04o, directory)\n", filename,
 	       mode);
 #endif
+#ifdef WIN32
+	if (mkdir(filename) == -1)
+#else
 	if (mkdir(filename, mode) == -1)
+#endif
 	{
 		if (errno == EEXIST)
 		{
@@ -515,7 +532,9 @@
 #ifdef DEBUG
 	printf("  ==> extracting: %s (fifo)\n", filename);
 #endif
+#ifndef WIN32
 	if (mkfifo(filename, mode) == -1)
+#endif
 	{
 #ifdef DEBUG
 		perror("mkfifo()");
